Algorithm.CSharp/StochasticIndicatorAndSubIndicatorsWarmUpRegressionAlgorithm.cs

```diff
@@ -0,0 +1,168 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data;
+using QuantConnect.Data.Consolidators;
+using QuantConnect.Indicators;
+using QuantConnect.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// This regression test ensures that the Stochastic indicator and its sub-indicators  
+    /// are properly initialized, warmed up, and returning meaningful values.  
+    /// It verifies that they do not return zero after warm-up.
+    /// </summary>
+    public class StochasticIndicatorAndSubIndicatorsWarmUpRegressionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private bool _dataPointsReceived;
+        private Symbol _spy;
+        private Stochastic _stochasticIndicator;
+        private Stochastic _stochasticHistory;
+
+        public override void Initialize()
+        {
+            SetStartDate(2020, 1, 1);
+            SetEndDate(2020, 2, 1);
+
+            _spy = AddEquity("SPY", Resolution.Hour).Symbol;
+
+            var dailyConsolidator = new TradeBarConsolidator(TimeSpan.FromDays(1));
+            _stochasticIndicator = new Stochastic("FIRST", 14, 3, 3);
+            RegisterIndicator(_spy, _stochasticIndicator, dailyConsolidator);
+
+            WarmUpIndicator(_spy, _stochasticIndicator, TimeSpan.FromDays(1));
+
+            _stochasticHistory = new Stochastic("SECOND", 14, 3, 3);
+            RegisterIndicator(_spy, _stochasticHistory, dailyConsolidator);
+
+            // Get historical data for warming up the stochasticHistory
+            var history = History(_spy, _stochasticHistory.WarmUpPeriod, Resolution.Daily);
```

Comment:
This test looks good. However as the bug actually involves the WarmUpPeriod attribute, if it is wrong then the check `_stochasticIndicator.StochK.Current.Value != _stochasticHistory.StochK.Current.Value` could still pass despite the indicators not being fully warmed up. An alternative may be to do a history request for a manually entered significantly larger value, such as 30 which should guarantee the indicator is warmed up.